crates/vault-registry/src/lib.rs

```diff
@@ -388,6 +392,55 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Atomically lock vault collateral in lending market and re-lock as vault collateral.
+        ///
+        /// Preconditions:
+        /// - the Vault is not liquidated
+        /// - the Vault does not have `to_be_issued` or `to_be_redeemed` tokens
+        /// - a market exists for the underlying currency
```

Comment:
- there is enough collateral s.t. the qtoken vault is above the secure threshold